pkg/controller/core/workload_controller.go

```diff
@@ -442,7 +449,7 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 		}
 		workload.SetDeactivationTarget(wl, kueue.WorkloadEvictedByAdmissionCheck, fmt.Sprintf("Admission check(s): %v, were rejected", stringsutils.Join(rejectedCheckNames, ",")))
 		if err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true, r.clock); err != nil {
-			return false, client.IgnoreNotFound(err)
+			return false, err
```

Comment:
Why do we need this change?
I have the same question in a couple of other places.